internal/provider/harness_k3s_resource.go

```diff
@@ -175,20 +179,50 @@ func (r *HarnessK3sResource) harness(ctx context.Context, data *HarnessK3sResour
 		kopts = append(kopts, k3s.WithImageRef(ref))
 	}
 
-	if !data.Sandbox.IsNull() {
-		sandbox := &HarnessK3sSandboxResourceModel{}
-		if diags := data.Sandbox.As(ctx, &sandbox, basetypes.ObjectAsOptions{}); diags.HasError() {
-			return nil, diags
+	if sandbox := data.Sandbox; sandbox != nil {
+		log.Info(ctx, "parsing sandbox", "raw", sandbox)
+
+		var ls []bundler.Layerer
+		for _, l := range sandbox.Layers {
+			ls = append(ls, bundler.NewFSLayer(
+				os.DirFS(l.Source.ValueString()),
+				l.Destination.ValueString(),
+			))
 		}
 
-		if !sandbox.Image.IsNull() {
+		var (
+			b   bundler.Bundler
+			err error
+		)
+		if sandbox.Image.ValueString() != "" {
 			ref, err := name.ParseReference(sandbox.Image.ValueString())
 			if err != nil {
 				return nil, []diag.Diagnostic{diag.NewErrorDiagnostic("invalid resource input", fmt.Sprintf("invalid sandbox image reference: %s", err))}
 			}
-			kopts = append(kopts, k3s.WithSandboxImageRef(ref))
+
+			b, err = bundler.NewAppender(ref)
+			if err != nil {
+				return nil, []diag.Diagnostic{diag.NewErrorDiagnostic("failed to create bundler", err.Error())}
+			}
+		} else {
+			b, err = bundler.NewApko(
+				bundler.ApkoWithRemoteOptions(r.store.ropts...),
```

Comment:
I think we might be able to use `remote.Reuse(pusher)` here which would side step the need to persist the remote options explicitly, I think